vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -130,4 +158,12 @@ def build(self):
         def callback_wrapper(trigger: None, **inputs: Dict[str, Any]) -> Dict[str, Any]:
             return self._action_callback_function(**inputs)
 
-        return action_components
+        # return action_components
+        return (
+            html.Div(
+                children=action_components,
+                id=f"{self.id}_action_model_components_div",
+            )
+            if action_components
+            else None
+        )
```

Comment:
Thanks for making this change - I think it's much better now it's consistent 👍 

Just one more thing to make it even more consistent: I don't think we have any other `build` methods that can return `None` (is that right @huong-li-nguyen?). So let's try to avoid it here also if we can:
```suggestion
        return html.Div(
                children=action_components,
                id=f"{self.id}_action_model_components_div",
            )
```
or possibly `html.Div(children=None)`, whatever works.

It means we have a redundant empty div, but if that doesn't cause any problems then I think this is preferable.

FYI the basis for these changes is polymorphism, which here basically means to maintain a single consistent contract (function signature and return types) across all implementations of `build`. Currently that (very informal) contract states that the return type is always a single Dash component and cannot be `None`. The simpler we can keep the contract (so no `Optional` or `Union` types), the better (e.g. for type-checking, fewer bugs, ease of understanding).

Reply:
Yes, that's right. Currently, every `build` method returns a component. However, what we do have in a few places is something like below where components of the children can return None instead of an `html.Div`. This is required such that the `html.Div` doesn't add unnecessary space, which could mess up our padding/margins. This should hopefully still be in line with what you mention above @antonymilne because all these inner components that could potentially return None are created within the return component, so shouldn't affect the type checks etc.

```
A = html.Div(children=[a_children]) if a_children or None
return html.Div(
                children=[a, b, c],
                id=f"{self.id}_action_model_components_div",
            )
```

@petar-qb , if you implement the above - can you double-check that that `html.Div `doesn't add any spacing? Given that this `html.Div` doesn't contain any visualizable components, it should be invisible (so no added spacings etc.). You might have to provide a className to the html.div called "action_components_container_outer" and then provide a width and height of 0?

Reply:
I've already checked it and it doesn't add any spacing because it doesn't contain any visible objects in it.